source/slang/hlsl.meta.slang

```diff
@@ -4553,31 +4562,32 @@ matrix<T, N, M> floor(matrix<T, N, M> x)
     MATRIX_MAP_UNARY(T, N, M, floor, x);
 }
 
-// Fused multiply-add for doubles
+// Fused multiply-add
+__generic<T : __BuiltinFloatingPointType>
 __target_intrinsic(hlsl)
 __target_intrinsic(glsl)
 __target_intrinsic(cuda, "$P_fma($0, $1, $2)")
 __target_intrinsic(cpp, "$P_fma($0, $1, $2)")
 __target_intrinsic(spirv, "OpExtInst resultType resultId glsl450 Fma _0 _1 _2")
 [__readNone]
-double fma(double a, double b, double c);
+T fma(T a, T b, T c);
```

Comment:
hlsl doesn't have `float`/`half` version of `fma`, should probably call `mad` instead.

Reply:
Only call `fma` when T is double.

Reply:
I see.. I didn't know that fma is only for double on HLSL.
I guess I didn't need to make the changes in diff.meta.slang after all.